src/main/java/edu/cornell/kfs/coa/businessobject/SubAccountGlobal.java

```diff
@@ -121,83 +126,63 @@ public SubAccountGlobal() {
-						if (!alreadyUpdatedIndexes.containsKey(i)) {
-							A21IndirectCostRecoveryAccount existingICR = a21IcrAccounts.get(i);
-							// check if we have a match on chart, account and percentage
-							if (existingICR.getIndirectCostRecoveryFinCoaCode().equalsIgnoreCase(newICR.getIndirectCostRecoveryFinCoaCode()) 
-									&& existingICR.getIndirectCostRecoveryAccountNumber().equalsIgnoreCase(newICR.getIndirectCostRecoveryAccountNumber()) 
-									&& existingICR.getAccountLinePercent().equals(newICR.getAccountLinePercent())) {
-								// set this to true if we have found a match
-								found = true;
-								// check if the they don't already both have the same active indicator
-								if (newICR.isActive() == existingICR.isActive()) {
-									// both the same, save position in temp and keep looking, if an entry exists that matches on chart, account and percentage but not same active indicator then we will update that one, otherwise we will consider a match on this temp entry
-									temp = i;
-								} else {
-									// done, stop looking and update the active indicator
-									existingICR.setActive(newICR.isActive());
-									alreadyUpdatedIndexes.put(i, i);
-									
-									// reset temp since we have found a better match
-									if(temp != -1){
-										temp = -1;
-									}
-								}
-							}
-						}
-						i++;
-					}
-
-					if (found && temp != -1) {
-						// no need to update but we will add the index in already updated since there was a match
-						alreadyUpdatedIndexes.put(temp, temp);
-					}
-
-					if (!found) {
-						// add to active add or inactive add list
-						A21IndirectCostRecoveryAccount icrAccount = createA21IndirectCostRecoveryAccountFromChange(subAccountGlobalDetail, newICR);
-						addList.add(icrAccount);
-
-					}
-
-				}
-				
-				List<A21IndirectCostRecoveryAccount> updatedA21IcrAccounts = new ArrayList<A21IndirectCostRecoveryAccount>();
-				updatedA21IcrAccounts.addAll(subAccountGlobalDetail.getSubAccount().getA21SubAccount().getA21IndirectCostRecoveryAccounts());
-				updatedA21IcrAccounts.addAll(addList);
+		    List<IndirectCostRecoveryAccount> icrAccounts = new ArrayList<IndirectCostRecoveryAccount>();
+		    
+		    for(IndirectCostRecoveryAccount icrAccount : subAccountGlobalDetail.getSubAccount().getA21SubAccount().getA21IndirectCostRecoveryAccounts()){
+		    	icrAccounts.add(icrAccount);
+		    }
 
-				subAccountGlobalDetail.getSubAccount().getA21SubAccount().setA21IndirectCostRecoveryAccounts(updatedA21IcrAccounts);
-			}
+		    updateIcrAccounts(subAccountGlobalDetail, icrAccounts);
 		    
 			changesToPersist.add(subAccount);
 		}
 		
 		return changesToPersist;
 	}
+	
+    public List<IndirectCostRecoveryAccountChange> getActiveIndirectCostRecoveryAccounts() {
+    	return SpringContext.getBean(GlobalObjectWithIndirectCostRecoveryAccountsService.class).getActiveIndirectCostRecoveryAccounts(this);
```

Comment:
One more SpringContext.getBean() call that could be replaced by a getter/setter.